strawberryfields/engine.py

```diff
@@ -609,22 +609,20 @@ def connection(self) -> xcc.Connection:
         )
 
     @property
-    def device_spec(self) -> DeviceSpec:
+    def device(self) -> Device:
         """The specification of the target device.
 
         Returns:
-            DeviceSpec: the device specification
+            ~strawberryfields.Device: the device specification
 
         Raises:
             requests.exceptions.RequestException: if there was an issue fetching
                 the device specifications from the Xanadu Cloud
         """
-        if self._spec is None:
-            target = self.target
-            connection = self.connection
-            device = xcc.Device(target=target, connection=connection)
-            self._spec = DeviceSpec(spec=device.specification)
-        return self._spec
+        if self._device is None:
+            device = xcc.Device(target=self.target, connection=self.connection)
+            self._device = Device(spec=device.specification, cert=device.certificate)
```

Comment:
Yeah we should definitely aim to merge them into a single abstraction, this is confusing!

At the moment, what is the main difference between the two? Is it that:

- `xcc.Device` is an object that represents the API contract with the Device, and is used for submitting and querying jobs?
- `sf.Device` stores the device specification and certificate?

Reply:
If that is the case, should the names reflect this difference?

Reply:
> `xcc.Device` is an object that represents the API contract with the Device

Correct! It is the canonical way to access any information about a device on the Xanadu Cloud, including device specifications, certificates, expected uptime, etc.

> and is used for submitting and querying jobs?

Nope. That's what `xcc.Job` is for. You only use `xcc.Device` for querying information about _devices_.

> `sf.Device` stores the device specification and certificate?

Yes, although it also has a few helper functions for interacting with `sf.Program` objects.

Reply:
There's actually not much of a difference between what the two classes contain. Apart from the specification and the certificate, `xcc.Device` contains a few the expected uptime and status of the device. So I actually think that them having the same name here makes sense (but we should probably merge them).

> xcc.Device [...] is used for submitting and querying jobs?

Not really, that's the `xcc.Job` class. The `xcc.Device` only fetches the specification, certificate and details dictionaries, and caches them.